src/hardware/driver/output.rs

```diff
@@ -0,0 +1,120 @@
+///! Driver output state handling.
+use crate::hardware::I2c1Proxy;
+use idsp::iir;
+use smlang::statemachine;
+
+use super::relay;
+
+pub struct Output {
+    ramp_iir: iir::IIR<f32>,
+}
+
+impl Output {
+    // 1 A/s current ramp-up
+    const RAMP_STEP: f32 = 1.0; // will be 1 mA current steps
+    const RAMP_DELAY: fugit::MillisDuration<u64> =
+        fugit::MillisDurationU64::millis(10); // will be current steps every 1 ms
+
+    pub fn new() -> Self {
+        Output {
+            ramp_iir: iir::IIR::new(0., -i16::MAX as _, i16::MAX as _), // Todo: sensible defaults for driver (this is for stabilizer now)
+        }
+    }
+}
+
+pub mod sm {
+    use super::*;
+    statemachine! {
+        transitions: {
+            *Disabled + Enable = EnableWaitRelay,
+            EnableWaitRelay + RelayDone = RampCurrent,
+            RampCurrent + CurrentStep / increment_current = RampCurrent,
```

Comment:
Can't you `RampCurrent + Enable = RampCurrent` and the same for `EnableWaitRelay + Enable`, and `DisableWaitRelay + Disable`, and `RampCurrent + Disable / reset_iir = DisableWaitRelay`?